pkg/kube/workload.go

```diff
@@ -514,3 +514,28 @@ func IsPodRunning(cli kubernetes.Interface, podName, podNamespace string) (bool,
 
 	return true, nil
 }
+
+func StatefulSetReplicas(ctx context.Context, kubeCli kubernetes.Interface, namespace, name string) (int32, error) {
+	sts, err := kubeCli.AppsV1().StatefulSets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return 0, errors.Wrapf(err, "Could not get STS{Namespace %s, Name: %s}, to figure out replicas", namespace, name)
+	}
+	return *sts.Spec.Replicas, nil
```

Comment:
Should we check if `sts.Spec.Replicas` is nil? What value do we get if workloads are not in `Running` state?

Reply:
I can add the check, but should that not be the responsibility of the function that is actually doing the scaling?
Also, even if the workload is not in running state, replicas count would be set.

```
k get pods -n default 
NAME                    READY   STATUS             RESTARTS   AGE
nginx-85f656985-pbdfs   0/1     ErrImagePull       0          10s
nginx-85f656985-rk8hn   0/1     ErrImagePull       0          10s
nginx-85f656985-svxcc   0/1     ImagePullBackOff   0          47s

k get deployments.apps -n default 
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   0/3     3            0           90s

k get deployments.apps -n default nginx -ojsonpath='{.spec.replicas}'
3
```

Reply:
Got it. It is a field of `spec`. Somehow I assumed that we are reading it from `status`. Thanks